### ISSUE TITLE
What flag does this flagpole fly?

### ISSUE THREAD

**Issue by 1ec5** [streetcomplete/StreetComplete#2705]
### General
Affected tag(s) to be modified/added: [`flag:type`](https://wiki.openstreetmap.org/wiki/Key:flag:type), [`flag:wikidata`](https://wiki.openstreetmap.org/wiki/Key:flag:wikidata), [`country`](https://wiki.openstreetmap.org/wiki/Key:country)
Question asked: **What flag does this flagpole fly?**

### Checklist
Checklist for quest suggestions (see guidelines):
- [x] 🚧 To be added tag is established and has a useful purpose
   * These keys are used thousands of times on flagpoles.
   * Some 3D renderers use these keys to color flags on flagpoles.
   * These keys make for [queries](https://wiki.openstreetmap.org/wiki/SPARQL_examples#Flags_by_frequency_and_type) that aren’t possible outside of OSM.
   * Identifying flags adds some variety to the StreetComplete user experience.
- [x] 🤔 Any answer the user can give must have an equivalent tagging (Quest should not reappear to other users when solved by one)
- [x] 🐿️ Easily answerable by everyone from the outside but a survey is necessary 
   * It’s usually impossible to identify a flag from aerial imagery.
   * A surveyor would be able to identify many flags easily with the help of visual aids.
- [x] 💤 Not an overwhelming percentage of elements have the same answer (No spam)
   * Some regions have a great variety of flags for various purposes; other regions only ever fly one flag.
- [x] 🕓 Applies to a reasonable number of elements (Worth the effort)
   * There are tens of thousands of `man_made=flagpole`s.

### Ideas for implementation

To ensure usability, the quest should present a gallery of flag images and allow searching them by name. Perhaps StreetComplete could [query Wikidata](https://query.wikidata.org/) for flag items and their Wikimedia Commons images. A menu of national and regional flags would be easier to implement based on local data (emoji, even), as long as the user’s location is geocoded first.

**Element selection:** 

`man_made=flagpole` with `flag:type=national` but no `country=*`
`man_made=flagpole` with another `flag:type=*` value but no `flag:wikidata=*`

**Metadata needed:** 

I’m told that, in some countries, flags rarely fly on most flagpoles. Maybe the quest can be disabled for those countries, but I think the default assumption would be that a flagpole does fly a flag on a regular basis.

**Proposed GUI:**

First select the flag type, then drill down to a more specific design:

* National
   * _Choose from a gallery of national flags, starting with the current country’s flag_
* Regional
   * _Choose from a gallery of state/provincial flags for the current country_
* Municipal
   * _[Search for municipal flags from Wikidata](https://w.wiki/39Rv)_
* Military
   * _Search for flags from Wikidata_
* Other governmental
   * _Search for flags from Wikidata_
* Commercial advertising
   * _Search for flags from Wikidata_
* Noncommercial organization
   * _Search for flags from Wikidata_
* Religious
   * _Search for flags from Wikidata_
* Athletic team
   * _Search for flags from Wikidata_
* Multiple flags
  * _Let the user go through the process of identifying a flag multiple times, from the top flag to the bottom flag_
* Don’t know (for example, because it’s a rainy day and no flag is flying)

Or if supporting Wikidata search would be too challenging, maybe a first version of the quest could focus on setting `flag:type=*` and only set `flag:wikidata=*` for national and regional flags, which are less open-ended.

**Comment by 1ec5** [comment]
> Quest should not reappear to other users when solved by one

A “None” option would make sense for a bare flagpole, but it looks like there isn’t an established way to explicitly tag that fact. `flag:type=none` is used only once, and `flag=no` is commonly used for something entirely different (to indicate that a bus stop doesn’t allow flagging down the bus driver).

There’s also the (hopefully somewhat rare) case that a flag does not have a Wikidata item yet. StreetComplete could offer to fill in [`flag:name`](https://wiki.openstreetmap.org/wiki/Key:flag:name) based on a freeform text input field. ([`name:etymology:wikidata:missing`](https://taginfo.openstreetmap.org/keys/name%3Aetymology%3Awikidata%3Amissing) is apparently a way to include disambiguating notes that aid in the creation of a Wikidata item, but I don’t think that’s as important for flag designs.)

**Comment by westnordost** [comment]
If you like mapping flags, I think you would get very happy in German allotments. Happy and confused, because they put up all kinds of flags. Football clubs, federal state flags, flags from other countries, pirate flags, fantasy flags, It's quite random.

![kleingartenverein-an-der-boye-in-bottrop-in-der-anlage-mit-rund-110-parzellen-haben-etliche-hobbygaertner-ihre-ganz-persoenlichen-fahnen-aufgehaengt-](https://user-images.githubusercontent.com/4661658/113215065-9b0e4080-927a-11eb-9f6f-6b19b9a0b9ad.jpg)

<img src="https://user-images.githubusercontent.com/4661658/113215063-9a75aa00-927a-11eb-85bd-f650986d0f59.jpg" width = 400 />
(Yes, I actually saw that one. There's no accounting for taste.)

There is actually an "official" flag for allotmentiers in Hamburg. (No, not official, there is nothing official about all these flags)
<img width=400 src="https://user-images.githubusercontent.com/4661658/113215397-13750180-927b-11eb-9d1d-a395b76b4da3.jpg"/>

Anyway, a bit off topic. I will come to the point in the next post.

**Comment by westnordost** [comment]
The quest would be problematic because of the following reasons:

> 🤔 Any answer the user can give must have an equivalent tagging

1. In Germany, many schools, government buildings etc have flag poles. But they don't usually fly a flag. Only on special occasions, flags are hoisted. So, a surveyor will not be able to tell which flag there is, and actually even if by chance there is on right now, because it's f.e. German Unification Day, there might be another one on some other day (f.e. the federal state flag, EU flag or whatnot). I reckon that the German and/or state flag is not often seen on public buildings is probably something unique to Germany, but it is clear that flag poles are made to make it exchange what flag to hoist.
2. you noted it yourself: No tag is documented to tag that there is no flag. And due to the possibly temporary nature of a flagpole having no flag, to add this sort of information would maybe even be wrong. (It would need to be "no flag right now")

> 🐿️ Easily answerable 

1. One can't assume that any user will know all the flags. So, your suggested UI won't work: What kind of flag is the second flag flag from the left on first picture in my last post? Is it a football club? Is it a federal state? A fantasy flag? No idea! So, a typical user would not even be able to select the category. What would be necessary to solve this properly would be some kind reverse image search that finds the flag when you make a photo.
2. StreetComplete is built to be offline-capable from the root. Once the data is downloaded, no quest type requires or should require a connection to the internet. So, qeuerying some wiki-data on demand is (currently) not possible. But that would be necessary, because the app should at least show the flag to the user as a picture before letting him/her make a selection.

> 🚧 To be added tag [...] has a useful purpose 

You mentioned 1. usage for 3D rendering, 2. "keys make for queries that aren't possible outside of OSM" and 3. it may be fun for the users to add.
1. 3D rendering does not count as a useful purpose. It should really be mentioned in the quest guidelines. Because with this argument, you can justify tagging anything really. It is literally tagging for the renderer.
2. Don't really know what you mean with that, the fact that you can query that data doesn't make it useful per sé.
3. I agree, flags are fun. And if the user even has the opportunity to learn about this flag by adding this information (thinking about allotment flags here), it might be very interesting. However, fun collecting an information doesn't really make it have a useful purpose.

So, in conclusion, I don't really see this quest in StreetComplete, at least not in the near future. The biggest blockers are:
1. no way to tag "no flag, at least right now"
2. needs reverse image search for flags

**Comment by 1ec5** [comment]
Thanks for entertaining this proposal and patiently explaining the constraints that make it infeasible. I am very tempted to map a German allotment garden now.
